src/main/kotlin/com/exactpro/th2/codec/openapi/writer/visitors/json/EncodeJsonObjectVisitor.kt

```diff
@@ -35,126 +36,145 @@ import com.fasterxml.jackson.databind.node.JsonNodeFactory
+import io.swagger.v3.oas.models.media.FileSchema
+import io.swagger.v3.oas.models.media.IntegerSchema
+import io.swagger.v3.oas.models.media.MapSchema
+import io.swagger.v3.oas.models.media.NumberSchema
+import io.swagger.v3.oas.models.media.PasswordSchema
 import io.swagger.v3.oas.models.media.Schema
+import io.swagger.v3.oas.models.media.StringSchema
+import io.swagger.v3.oas.models.media.UUIDSchema
 import java.math.BigDecimal
-
-
-class EncodeJsonObjectVisitor(override val from: Message) : EncodeVisitor<Message>() {
-
-    private val rootNode: ObjectNode = mapper.createObjectNode()
-
-    override fun visit(fieldName: String, defaultValue: Schema<*>?, fldStruct: Schema<*>, required: Boolean, schemaWriter: SchemaWriter) {
-        from.getField(fieldName, required)?.getMessage()?.let { nextMessage ->
-            val visitor = EncodeJsonObjectVisitor(nextMessage)
-            schemaWriter.traverse(visitor, fldStruct)
+import java.time.OffsetDateTime
+
+open class EncodeJsonObjectVisitor(override val from: Message, override val settings: VisitorSettings) : SchemaVisitor.EncodeVisitor<Message>() {
+    internal val rootNode: ObjectNode = mapper.createObjectNode()
+
+    override fun visit(fieldName: String, fldStruct: Schema<*>, required: Boolean, throwUndefined: Boolean) {
+        from.getField(fieldName, required)?.let { field ->
+            when {
+                field.kindCase.number == 1 -> {
+                    rootNode.set<ObjectNode>(fieldName, ObjectNode(mapper.nodeFactory))
+                    return
+                }
+                !field.hasMessageValue() -> error("$fieldName is not an message: ${field.kindCase}")
+            }
+            val message = field.getMessage()!!
+            val visitor = EncodeJsonObjectVisitor(message, settings)
+            val writer = SchemaWriter(settings.openAPI)
+            writer.traverse(visitor, fldStruct, throwUndefined)
             rootNode.set<ObjectNode>(fieldName, visitor.rootNode)
-        }
+        } ?: fldStruct.default?.let { error("Default values isn't supported for objects") }
     }
 
-    override fun visit(fieldName: String, defaultValue: String?, fldStruct: Schema<*>, required: Boolean) {
-        from.getField(fieldName, required)?.getString()?.let { value ->
-            fldStruct.checkEnum(value, fieldName)
-            rootNode.put(fieldName, value)
-        } ?: defaultValue?.let {
-            rootNode.put(fieldName, defaultValue)
-        }
+    override fun visit(fieldName: String, fldStruct: ArraySchema, required: Boolean, throwUndefined: Boolean) {
+        val itemSchema = settings.openAPI.getEndPoint(fldStruct.items)
+
+        from.getField(fieldName, required)?.let { field ->
+            when {
+                field.kindCase.number == 1 -> {
+                    rootNode.putArray(fieldName)
+                    return
+                }
+                !field.hasListValue() -> error("$fieldName is not an list: ${field.kindCase}")
+            }
```

Comment:
```suggestion
            if (field.kindCase == NULL_VALUE) {
                rootNode.putArray(fieldName)
                return
            }
            
            check(field.hasListValue()) { "$fieldName is not a list: ${field.kindCase}" }
```

> field.kindCase == NULL_VALUE

Looks like we need an utility function for this in common-lib